pkg/users/accounts-list.js

```diff
@@ -18,72 +18,281 @@
+            ),
+            props: { width: 20 },
+        },
+        {
+            title: <UserActions account={account} />,
+            props: { className: "pf-c-table__action" }
+        },
+    ];
 
-export function AccountsList({ accounts, current_user }) {
-    const filtered_accounts = accounts.filter(function(account) {
-        return !((account.uid < 1000 && account.uid !== 0) ||
-                 account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
-                 account.shell === '/bin/false');
+    return { columns };
+};
+
+const mapGroupsToAccount = (accounts, groups) => {
+    return accounts.map(account => {
+        const accountGroups = [];
+        groups.forEach(group => {
+            if (group.userlist.find(accountName => accountName === account.name))
+                accountGroups.push(group.name);
+        });
+        account.groups = accountGroups;
+
+        return account;
     });
+};
+
+const AccountsList = ({ accountsInfo, current_user, groups }) => {
+    const [currentTextFilter, setCurrentTextFilter] = useState("");
+
+    const accounts = mapGroupsToAccount(accountsInfo, groups);
 
-    filtered_accounts.sort(function (a, b) {
-        if (current_user === a.name) return -1;
-        else if (current_user === b.name) return 1;
-        else if (!a.gecos) return -1;
-        else if (!b.gecos) return 1;
-        else return a.gecos.localeCompare(b.gecos);
+    const filtered_accounts = accounts.filter(account => {
+        if ((account.uid < 1000 && account.uid !== 0) ||
+                 account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
+                 account.shell === '/bin/false')
+            return false;
+
+        if (currentTextFilter !== "" &&
+            (account.name.toLowerCase().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+            (account.gecos.toLowerCase().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+            (account.uid.toString().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+            (!account.groups.find(group => group.toLowerCase().indexOf(currentTextFilter.toLowerCase()) !== -1)))
+            return false;
+
+        return true;
     });
 
+    const columns = [
+        { title: _("Username"), sortable: true },
+        { title: _("Full name"), sortable: true },
+        { title: _("ID"), sortable: true },
+        { title: _("Last active"), sortable: true },
+        { title: _("Group") },
+    ];
+
+    const sortRows = (rows, direction, idx) => {
+        const sortedRows = rows.sort((a, b) => {
+            const aitem = a.columns[idx];
+            const bitem = b.columns[idx];
+            const aname = a.columns[0];
+            const bname = b.columns[0];
+
+            // current user is always first
+            if (aname.sortKey === current_user)
+                return direction === SortByDirection.asc ? -1 : 1;
+            if (bname.sortKey === current_user)
+                return direction === SortByDirection.asc ? 1 : -1;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17830-20221130-151606-eb9be7bf-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)